File: SmallKatas/PerfectNumbers/Ruby/PerfectNumberCalculatorTest.rb
require "test/unit"
require "./PerfectNumberCalculator"

class TestPerfectNumbers < Test::Unit::TestCase
    def testFourIsNotAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(4)

        # Assert
        assert_equal(false, actual, "4 is not a perfect number")
    end

    def testSixIsAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(6)

        # Assert
        assert(actual, "6 is a perfect number")
    end

    def testTenIsNotAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(10)

        #Assert
        assert_equal(false,actual, "10 is not a perfect number")
    end

    def testTwentyIsNotAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(20)

        # Assert
        assert_equal(false,actual, "20 is not a perfect number")
    end

    def testTwentyEightIsAPerfectNumber()
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(28)

        # Assert
        assert(actual, "28 is a perfect number")
    end

    def testOneHundredTwelveIsNotAPerfectNumber
    
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(112)

        # Assert
        assert_equal(false, actual, "112 is not a perfect number")
    end

    def testFourhundredNinetySixIsAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(496)

        # Assert
        assert(actual, "496 is a perfect number")
    end

    def testFourThousandOneHundredTwentyNineIsNotAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(4129)

        # Assert
        assert_equal(false,actual, "4129 is not a perfect number")
    end

    def testEightThousandOneHundredTwentyEightIsAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(8128)

        # Assert
        assert(actual, "8128 is a perfect number")
    end

    def testTwelveThousandFourHundredElevenIsNotAPerfectNumber
        # Arrange
        perfectNumberCalculator = PerfectNumberCalculator.new

        # Act
        actual = perfectNumberCalculator.IsPerfectNumber(12411)

        # Assert
        assert_equal(false,actual, "12411 is not a perfect number")
    end
end

File: SmallKatas/PerfectNumbers/Java/PerfectNumberCalculatorTest.java
import junit.framework.TestCase;

public class PerfectNumberCalculatorTest extends TestCase {
    private boolean getActual(int number){
        return new PerfectNumberCalculator().isPerfectNumber(number);
    }

    public void testFourIsNotAPerfectNumber()
    {
        assertEquals("4 is not a perfect number",false,getActual(4));
    }

    public void testSixIsAPerfectNumber()
    {
        assertEquals("6 is a perfect number",true,getActual(6));
    }

    public void testTenIsNotAPerfectNumber()
    {
        assertEquals("10 is not a perfect number",false,getActual(10));
    }

    public void testTwentyIsNotAPerfectNumber()
    {
        assertEquals("20 is not a perfect number",false,getActual(20));
    }

    public void testTwentyEightIsAPerfectNumber()
    {
        assertEquals("28 is a perfect number",true, getActual(28));
    }

    public void testOneHundredTwelveIsNotAPerfectNumber()
    {
      assertEquals("112 is not a perfect number",false, getActual(112));
    }

    public void testFourhundredNinetySixIsAPerfectNumber()
    {
        assertEquals("496 is a perfect number",true,getActual(496));
    }

    public void testFourThousandOneHundredTwentyNineIsNotAPerfectNumber()
    {
        assertEquals("4129 is not a perfect number",false,getActual(4129));
    }

    public void testEightThousandOneHundredTwentyEightIsAPerfectNumber()
    {
        assertEquals("8128 is a perfect number",true,getActual(8128));
    }

    public void testTwelveThousandFourHundredElevenIsNotAPerfectNumber()
    {
      assertEquals("12411 is not a perfect number",false,getActual(12411));
    }
}

